janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -103,30 +104,12 @@ public ElasticMajorVersion getMajorVersion() {
             return majorVersion;
         }
 
-        final Pattern pattern = Pattern.compile("(\\d+)\\.\\d+\\.\\d+");
         majorVersion = DEFAULT_VERSION;
         try {
             final Response response = delegate.performRequest(REQUEST_TYPE_GET, REQUEST_SEPARATOR);
             try (final InputStream inputStream = response.getEntity().getContent()) {
                 final ClusterInfo info = mapper.readValue(inputStream, ClusterInfo.class);
-                final String version = info.getVersion() != null ? (String) info.getVersion().get("number") : null;
-                final Matcher m = version != null ? pattern.matcher(version) : null;
-                switch (m != null && m.find() ? Integer.valueOf(m.group(1)) : -1) {
-                    case 1:
-                        majorVersion = ElasticMajorVersion.ONE;
-                        break;
-                    case 2:
-                        majorVersion = ElasticMajorVersion.TWO;
-                        break;
-                    case 5:
-                        majorVersion = ElasticMajorVersion.FIVE;
-                        break;
-                    case 6:
-                        majorVersion = ElasticMajorVersion.SIX;
-                        break;
-                    default:
-                        throw new IllegalArgumentException("Unsupported Elasticsearch server version: " + version);
-                }
+                return ElasticMajorVersion.parse(info.getVersion() != null ? (String) info.getVersion().get("number") : null);
```

Comment:
`majorVersion = ElasticMajorVersion.parse(...);` so it only makes one server call for this during application lifetime.